src/python/WMComponent/TaskArchiver/CleanCouchPoller.py

```diff
@@ -221,3 +227,26 @@ def cleanAllLocalCouchDB(self, workflowName):
         # other wise return True.
         return True
         
+    def cleanAlreadyArchivedWorkflows(self):
+        """
+        loop through the workflows in couchdb, if archived delete all the data in couchdb
+        """
+        
+        localWMStats = self.wmstatsCouchDB.getDBInstance()
+        options = {"group_level": 1, "reduce": True}
+        try:
+            results = localWMStats.loadView("WMStatsAgent", "allWorkflows", options = options)['rows']
+            requestNames = [x['key'] for x in results]
+        except Exception as ex:
+            errorMsg = "Error on loading workflow lists %s" % requestNames
```

Comment:
I noticed in my tests that it was trying to use requestNames before assignment (because I mistakenly applied the patch - couch part :-(). So you may remove it from the except block